### PR TITLE
Projection coordinate lookup

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

**Comment by jdroenner** [comment]
[Koordinaten in der Projektion erst checken, dann umprojezieren.](https://trello.com/c/UzBihqEA/394-koordinaten-in-der-projektion-erst-checken-dann-umprojezieren)

File: operators/src/adapters/raster_subquery_adapter.rs [hunk 1]
```diff
@@ -611,19 +611,54 @@ where
...
+            // fastpath to skip filter
+            vec![]
+        };
+
+        // unzip to use batch projection
+        let (idxs, coords): (Vec<_>, Vec<_>) = idx_coords.into_iter().unzip();
+
+        // project
         let proj = CoordinateProjector::from_known_srs(self.out_srs, self.in_srs)?;
         let projected_coords = project_coordinates_fail_tolerant(&coords, &proj);
```
Comment (michaelmattig) [inline operators/src/adapters/raster_subquery_adapter.rs:659]: does it still have to be fail tolerant?

**Review by michaelmattig — CHANGES_REQUESTED** [review]

File: operators/src/adapters/raster_subquery_adapter.rs [hunk 2]
```diff
@@ -611,19 +611,54 @@ where
         let output_raster =
             EmptyGrid::new(tile_info.tile_size_in_pixels, self.no_data_and_fill_value);
 
+        // generate a projector which transforms wgs84 into the projection we want to produce.
+        let valid_bounds_proj =
```
[the remark this answers is not in the document]
Comment (jdroenner) [reply operators/src/adapters/raster_subquery_adapter.rs]: d5f3319

File: operators/src/adapters/raster_subquery_adapter.rs [hunk 3]
```diff
@@ -611,19 +611,54 @@ where
...
+            CoordinateProjector::from_known_srs(SpatialReference::epsg_4326(), self.out_srs)?;
+
+        // transform the bounds of the input srs (coordinates are in wgs84) into the output projection.
+        // TODO check if  there is a better / smarter way to check if the coordinates are valid.
+        let valid_bounds = self
```
[the remark this answers is not in the document]
Comment (jdroenner) [reply operators/src/adapters/raster_subquery_adapter.rs]: d5f3319

File: operators/src/adapters/raster_subquery_adapter.rs (hunk 1 above)
Comment (jdroenner) [reply operators/src/adapters/raster_subquery_adapter.rs]: d5f3319

**Comment by jdroenner** [comment]
redesigned the bounds check and generate valid bounds only once

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**